src/jsrm/utils/lie_operators.py

```diff
@@ -0,0 +1,271 @@
+import jax
+import jax.numpy as jnp
+from jax import jit
+
+# for documentation
+from jax import Array
+from typing import Sequence
+
+@jit
+def tilde_SE3( 
+    vec3:Array
+)-> Array:
+    """
+    Computes the tilde operator of SE(3) for a 3D vector.
+    
+    Args:
+        vec (Array): array-like, shape (3,1)
+            A 3-dimensional vector.
+    
+    Returns:
+        Array: shape (3, 3)
+            A 3x3 matrix representing the tilde operator of the input vector.
+    """
+    vec3 = vec3.reshape(-1)  # Ensure vec3 is a 1D array
+    if vec3.shape != (3,):
+        raise ValueError("Input vector must be of shape (3,)")
+    
+    # Extract components of the vector
+    x, y, z = vec3.flatten()
+    
+    # Use JAX's array creation for better performance
+    Mtilde = jnp.array([
+        [0, -z, y],
+        [z, 0, -x],
+        [-y, x, 0]
+    ])
+    return Mtilde
+
+# def adjoint_SE2(
+#     vec3:Array
+# )-> Array:
+#     """
+#     Computes the adjoint representation of a vector of se(2).
+
+#     Args:
+#         vec (Array): array-like, shape (3,1)
+
+#     Returns:
+#         Array: shape (3, 3)
+#             A 3x3 matrix representing the adjoint transformation of the input screw vector.
+#     """
+#     return tilde_SE3(vec3)  # The adjoint representation for se(2) is the same as the tilde operator
+
+@jit
+def adjoint_SE3(
+    vec6:Array
+)-> Array:
+    """
+    Computes the adjoint representation of a vector of se(3).
+
+    Args:
+        vec (Array): array-like, shape (6,1)
+            A 6-dimensional vector representing the screw. 
+            The first three elements correspond to the angular component, 
+            and the last three elements correspond to the linear component.
+
+    Returns:
+        Array: shape (6, 6)
+            A 6x6 matrix representing the adjoint transformation of the input screw vector.
+    """
+    vec6 = vec6.reshape(-1) # Ensure vec6 is a 1D array
+    if vec6.shape != (6,):
+        raise ValueError("Input vector must be of shape (6,)")
+    
+    ang = vec6[:3].reshape((3, 1))  # Angular as a (3,1) vector
+    lin = vec6[3:].reshape((3, 1))  # Linear as a (3,1) vector
+    
+    angtilde = tilde_SE3(ang)  # Tilde operator for angular part
+    lintilde = tilde_SE3(lin)  # Tilde operator for linear part
+    
+    adj = jnp.zeros((6, 6))
+    adj = adj.at[:3, :3].set(angtilde)  # Fill the top-left 3x3 block with the tilde operator of the angular part
+    adj = adj.at[3:, 3:].set(angtilde)  # Fill the bottom-right 3x3 block with the tilde operator of the angular part
+    adj = adj.at[3:, :3].set(lintilde)  # Fill the bottom-left 3x3 block with the tilde operator of the linear part
```

Comment:
Please use `jnp.block` (https://docs.jax.dev/en/latest/_autosummary/jax.numpy.block.html)  instead of `.at[].set()`